### PR TITLE
[API] Fix artifact storing without project in body

### PR DESCRIPTION
**Pull request by Tankilevitch**
https://jira.iguazeng.com/browse/ML-1322

### CODE REVIEW

File: tests/integration/sdk_api/artifacts/test_artifacts.py [hunk 1]
```diff
@@ -55,3 +55,37 @@ def test_list_artifacts_filter_by_kind(self):
             project=prj, category=mlrun.api.schemas.ArtifactCategories.dataset
         )
         assert len(artifacts) == 1, "bad number of dataset artifacts"
+
+    def test_store_artifact_with_empty_dict(self):
```
Comment (Hedingber) [inline tests/integration/sdk_api/artifacts/test_artifacts.py:59]: We should always strive for having a functionality being tested by the fastest possible test:
* **System** tests are the slowest one - in order to run a test correctly you need to build a whole version out of the code, transplant it in the system etc..
* **Integration** tests are faster - in order to run a test you (done automatically when you run it) build a docker image of the API, run it, and do actions in the client
* **Unit** tests are the fastest - all you need to run them is to install all requirements

Your change here is in the API code, client is irrelevant, therefore the correct place IMO to test it is the API tests, which are unit tests that bring up a mock API inside the test process

File: mlrun/api/crud/artifacts.py [hunk 2]
```diff
@@ -23,6 +23,15 @@ def store_artifact(
         project: str = mlrun.mlconf.default_project,
     ):
         project = project or mlrun.mlconf.default_project
+        # In case project is an empty string the setdefault won't catch it
+        if not data.setdefault("project", project):
+            data["project"] = project
```
Comment (Hedingber) [inline mlrun/api/crud/artifacts.py:28]: If you handled the case of an empty string, I suggest that you'll cover it in the tests (maybe I missed it ?)

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/api/crud/artifacts.py (hunk 2 above)
Comment (Tankilevitch) [reply mlrun/api/crud/artifacts.py:28]: added unittests

File: tests/integration/sdk_api/artifacts/test_artifacts.py (hunk 1 above)
Comment (Tankilevitch) [reply tests/integration/sdk_api/artifacts/test_artifacts.py:59]: So you suggesting removing those tests?

**Comment by Hedingber** [comment]
I meant that you'll remove the integration tests, other than that LGTM

**Review by Hedingber — APPROVED** [review]

**Pull request merged**